actors/builtin/miner/monies.go

```diff
@@ -44,9 +44,16 @@ func PledgePenaltyForUndeclaredFault(epochTargetReward abi.TokenAmount, networkQ
 }
 
 // Penalty to locked pledge collateral for the termination of a sector before scheduled expiry.
-func pledgePenaltyForTermination(s *SectorOnChainInfo, epochTargetReward abi.TokenAmount, networkQAPower abi.StoragePower, qaSectorPower abi.StoragePower) abi.TokenAmount {
-	//TODO #437 compute correct termination fee using initial pledge value from SectorOnChainInfo
-	return big.Zero() // PARAM_FINISH
+func PledgePenaltyForTermination(s *SectorOnChainInfo, currEpoch abi.ChainEpoch, epochTargetReward abi.TokenAmount, networkQAPower abi.StoragePower, qaSectorPower abi.StoragePower) abi.TokenAmount {
+	// max(SP(t), InitialPledge + BR(StartEpoch)*min(SectorAgeInDays, 180)) where BF(StartEpoch)=InitialPledge/InitialPledgeFactor
+	sectorAge := big.NewInt(int64(minEpoch(currEpoch-s.Activation, 180*builtin.EpochsInDay)))
+	return big.Max(
+		PledgePenaltyForUndeclaredFault(epochTargetReward, networkQAPower, qaSectorPower),
+		big.Add(
+			InitialPledgeFactor,
```

Comment:
From the comment above I would have thought this would be `s.InitialPledge`, not `InitialPledgeFactor`

Reply:
I have been meaning to add unit tests specifically for all these calculation functions. Let's start with this one.

Could you add some basic tests outlining the calculation results? Especially useful might be border cases like 1 epoch, 180days, >180days, and some right-order-of-magnitude numbers for pledge, power etc.